app/controllers/documents_controller.rb

```diff
@@ -37,7 +52,17 @@ def destroy
   private
 
   def permitted_params
-    params.require(:document).permit(:category, :file)
+    params.require(:document).permit(:category, :file, :file_name)
+  end
+
+  def update_transaction
+    @update_transaction ||=
+      begin
+        Documents::UpdateTransaction.new.with_step_args(
+          change_filename: [document: @document],
+          update_file: [document: @document]
+        ).call(permitted_params)
+      end
```

Comment:
* Why did you need to memoize this ?
* I don't find it very useful to use `step_args` here, as you pass the same argument to each steps, and the value of these arguments are also the same. I would prefer if you just use `.new.call(permitted_params.merge(document: @document))`, as that would make it obvious that the transaction will work with the same document